docs: add a contributing FAQs document (#5352)
stdlib-js/stdlib: docs/contributing/setting_up_a_devcontainer.md

<!--

@license Apache-2.0

Copyright (c) 2025 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# Devcontainer Setup

> > Step-by-Step Tutorial on How to Set Up the Repository in a Dev Container

Reviewer: ```suggestion
> Step-by-step tutorial on how to set up a stdlib development environment in a dev container.
```

## Introduction

We appreciate your interest in contributing to stdlib! Below, we've provided a step-by-step tutorial on how to set up the project locally on your device using a dev container.

> The stdlib repository includes a preconfigured dev container, making it the easiest way to set up your development environment. It ensures proper linting, EditorConfig, and tooling are configured right from the start.

Reviewer: We should probably describe what, exactly, a dev container issue. A few sentences should suffice.

**Note:** The dev container does not yet support ARM64 architectures. For more information, or if you're interested in adding ARM64 support, you can visit this [issue][devcontainer-issue].

### Prerequisites

Setting up the stdlib dev container **requires** the following prerequisites:

-   [Git][git]: Version control
-   [Docker][docker]: Containerization
-   [VS Code][vscode]: Preferred IDE

### Download

To acquire the source code, first navigate to the parent directory where you want to place the project’s [Git][git] repository.

<!-- run-disable -->

```bash
$ cd /path/to/parent/destination/directory
```

Next, clone the repository.

<!-- run-disable -->

```bash
$ git clone https://github.com/stdlib-js/stdlib.git
```

If you are wanting to contribute to stdlib, first [fork][github-fork] the repository and amend the previous command.

<!-- run-disable -->

```bash
$ git clone https://github.com/<username>/stdlib.git
```

Open the repository in VS Code.

```bash
$ cd stdlib && code .
```

When prompted, open the repository in the dev container.

<div class="image" align="center">

>     <a title="Step 1">

Reviewer: You can remove these anchor tags, as you aren't using them in this document.

>         <img width=“480” src="./img/step1.png" alt="Dev container setup tutorial">

Reviewer: ```suggestion
        <img width="480" src="./img/step1.png" alt="Dev container setup tutorial">
```

>         <img width="480" src="./img/step1.png" alt="Dev container setup tutorial">

Reviewer: These alt-text description should be more descriptive. For those with screen readers, every image will read as if they are the same thing.

Reviewer: See, e.g., https://accessibility.huit.harvard.edu/describe-content-images

    </a>
    <br>
</div>

Please be patient, as the post-create script may take some time to install all the required languages and dependencies.

<div class="image" align="center">
    <a title="Step 2">

>         <img width=“480” src="./img/step2.png" alt="Dev container setup tutorial">

Reviewer: You need to remove fancy quotes from each of the images below.

```suggestion
        <img width="480" src="./img/step2.png" alt="Dev container setup tutorial">
```

Author: on it
